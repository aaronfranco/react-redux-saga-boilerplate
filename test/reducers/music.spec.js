import reducer from 'reducers/music';
import { ActionTypes } from 'constants/index';

describe('Music', () => {
  it('should return the initial state', () => {
    expect(reducer.music(undefined, {}))
      .toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_JANE_REQUEST}`, () => {
    expect(reducer.music(undefined, {
      type: ActionTypes.GET_JANE_REQUEST,
      payload: {  },
    }))
      .toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_JANE_SUCCESS}`, () => {
    expect(reducer.music(undefined, {
      type: ActionTypes.GET_JANE_SUCCESS,
      payload: {},
    }))
      .toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_JANE_FAILURE}`, () => {
    expect(reducer.music(undefined, {
      type: ActionTypes.GET_JANE_FAILURE,
      payload: {},
    }))
      .toMatchSnapshot();
  });
});
