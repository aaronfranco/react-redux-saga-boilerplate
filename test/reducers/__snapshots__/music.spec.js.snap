// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music should handle GET_JANE_FAILURE 1`] = `
Object {
  "message": "Something went wrong",
  "music": Array [],
  "name": "",
  "state": "error",
  "username": "",
}
`;

exports[`Music should handle GET_JANE_REQUEST 1`] = `
Object {
  "message": "",
  "music": undefined,
  "name": undefined,
  "state": "running",
  "username": undefined,
}
`;

exports[`Music should handle GET_JANE_SUCCESS 1`] = `
Object {
  "message": "",
  "music": undefined,
  "name": undefined,
  "state": "loaded",
  "username": undefined,
}
`;

exports[`Music should return the initial state 1`] = `
Object {
  "message": "",
  "music": Array [],
  "name": "",
  "state": "idle",
  "username": "",
}
`;
